### DISCUSSION TITLE
How to trigger Fastify server reload in dev mode when a shared monorepo library file is changed?

### DISCUSSION THREAD

**Discussion by uncvrd** [fastify/fastify#5250] [Q&A]
Hi all - quick question. I have a shared library in a pnpm monorepo (like `/packages/trpc`) that I'd like to use in my fastify app (like `/apps/backend`). When I make a change in this package I use tsup to go from ts -> cjs/esm. I'd like to have my fastify app restart while in dev watch mode when those changes are made too. However, since this shared package is a node_module it is (i think) being ignored by fastify cli's watch mode. 

Has anybody come up with a good dev mode experience when modifying shared libraries? It's a little inconvenient to have to force the server to restart so that my package changes are reflected in the server. Anyone have any thoughts on this? Thank a lot!

Here's my current dev mode script (nothing too far from the default fastify-cli script besides tsc-alias):

`tsc && concurrently -k -p \"[{name}]\" -n \"TypeScript,TypeScript Alias,App\" -c \"yellow.bold,red.bold,cyan.bold\" \"tsc -w --preserveWatchOutput\" \"tsc-alias -w\" \"fastify start --options --ignore-watch='.ts$' -w -l info -P dist/app.js\"`

---

**TL;DR**

Is it possible to have Fastify detect file changes from specific node_modules (such as a local shared monorepo package) and reload accordingly?

**Comment by metcoder95** [comment]
Hmm, `nodemon` watching at those `node_modules` path does not work?
